spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitGatewaySupport.java

```diff
@@ -79,18 +80,39 @@ public final ConnectionFactory getConnectionFactory() {
 	 * Set the RabbitTemplate for the gateway.
 	 * @param rabbitTemplate The Rabbit template.
 	 * @see #setConnectionFactory(org.springframework.amqp.rabbit.connection.ConnectionFactory)
+	 * @deprecated - use {@link #setRabbitOperations(RabbitOperations)}
 	 */
+	@Deprecated
 	public final void setRabbitTemplate(RabbitTemplate rabbitTemplate) {
 		this.rabbitTemplate = rabbitTemplate;
 	}
 
 	/**
+	 * Set the RabbitTemplate for the gateway.
+	 * @param rabbitTemplate The Rabbit template.
+	 * @see #setConnectionFactory(org.springframework.amqp.rabbit.connection.ConnectionFactory)
+	 */
+	public final void setRabbitOperations(RabbitOperations rabbitTemplate) {
+		Assert.isInstanceOf(RabbitTemplate.class, rabbitTemplate);
```

Comment:
Seems for me the original concern that `RabbitTemplate` maybe AOP proxy, therefore with assert is gonna fail.
Isn't it ?